content/academy/anti_scraping.md

```diff
@@ -0,0 +1,76 @@
+
+Once one of these methods detects that the entity is a bot, it will take countermeasures depending on how advanced its techniques are.
+
+A common workflow of a website after it has detected a bot goes as follows:
+
+1. The bot is added to the "greylist" (a list of suspicious IP addresses, or any other value that can be used to uniquely identify the bot).
+2. A [Turing test](https://en.wikipedia.org/wiki/Turing_test) is provided to the bot (captcha). If the bot succeeds, it is added to the whitelist.
+3. If the captcha is failed, the bot is added to the blacklist.
+
+One thing to keep in mind while navigating through this course is that dvanced scraping methods are able to identify non-humans not only by one value (such as a single header value, or IP address), but are able to identify them through more complex things such as header combinations.
+
+## [](#common-measures) Common Anti-Scraping Measures
+
+### 1. Browser fingerprinting
+
+One of the most successful methods is collecting the browser's "fingerprint", which is a fancy name for information such as fonts, audio codecs, canvas fingerprint, graphics card, and more. Browser fingerprints are highly unique, so they are a reliable means of identifying a specific user (or bot). If the fingerprint provides different/inconsistent information, the entity is added to the greylist.
+
+> It's important to note that this method also blocks all entities that cannot evaluate JavaScript (such as bots sending only static HTTP requests), and combines both of the fundamental methods mentioned earlier.
+
+### 2. Regular structure changes
+
+By definition, this is not an anti-scraping method, but it can heavily affect the reliability of a scraper. If your target website drastically changes its CSS selectors, and your scraper is heavily reliant on selectors, it could break. In principle, websites using this method change their HTML structure or CSS selectors randomly and frequently, making the parsing of the data harder, and requiring more maintenance of the bot.
+
+### 3. Interval analysis
+
+This technique is based on analyzing the time intervals of the visit of a website. If the times are very similar, the entity is added to the greylist. This method’s premise is that the bot runs in regular intervals by, for example, a CRON job that starts every Monday. It is a long-term strategy, so it should be used as an extension. This technique needs only the information from the HTTP request to identify the frequency of the visits.
+
+### 4. URL analysis
+
+Solely based on the way how the bots operate. It comperes data-rich pages visits and the other pages visits. The ratio of the data-rich and regular pages has to be high to identify the bot and reduce false positives successfully.
```

Comment:
I know this is from Notion, but could you rewrite it a bit please? It's not easy to understand what this actually means.